src/plugins/intel_cpu/src/nodes/scatter_update.cpp

```diff
@@ -263,6 +261,427 @@ static std::vector<size_t> getBlockND(const VectorDims& shape) {
+                for (size_t worker = start; worker < end; worker++) {
+                    auto indices_offset = *ptr_indices_offset + idx * indicesBlock_axisplus1;
+                    int64_t idxValue =  getIndicesValue(indicesPtr, indices_offset);
+                    if (idxValue < 0) idxValue += data_dim_size;
+                    ASSERT_DEBUG_ONLY(idxValue < data_dim_size && idxValue >= 0, "invalid index value.");
+                    auto dst = &dataPtr[ptr_dst_offset[0] + idxValue * dataBlock_axisplus1];
+                    auto src = &updatePtr[indices_offset];
+                    kernel_func(dst, src);
+                    mean_reduction_counters[dst] += 1;
+                    ptr_indices_offset++;
+                    ptr_dst_offset++;
+                }
+            }
+
+            // average
+            for (const auto& counter : mean_reduction_counters) {
+                auto dst = counter.first;
+                const auto N = counter.second + static_cast<int32_t>(use_init_val);
+                *dst = static_cast<DataType>(static_cast<double>(*dst) / N);
+            }
+        }
+    });
+}
+
+template <typename DT>
+void ScatterUpdate::scatterElementsUpdate_dispatch(const MemoryPtr& dstMemPtr, const MemoryPtr& indicesMemPtr, const MemoryPtr& updateMemPtr,
+                        int axis) {
+    switch (reduction_type) {
+    case ScatterUpdate::Reduction::NONE :
+        scatterElementsUpdate<DT>(dstMemPtr, indicesMemPtr, updateMemPtr, axis, data_assign);
+        break;
+    case ScatterUpdate::Reduction::SUM :
+        scatterElementsUpdate<DT>(dstMemPtr, indicesMemPtr, updateMemPtr, axis, reduce_add);
+        break;
+    case ScatterUpdate::Reduction::MAX :
+        scatterElementsUpdate<DT>(dstMemPtr, indicesMemPtr, updateMemPtr, axis, reduce_maximum);
+        break;
+    case ScatterUpdate::Reduction::MIN :
+        scatterElementsUpdate<DT>(dstMemPtr, indicesMemPtr, updateMemPtr, axis, reduce_minimum);
+        break;
+    case ScatterUpdate::Reduction::PROD:
+        scatterElementsUpdate<DT>(dstMemPtr, indicesMemPtr, updateMemPtr, axis, reduce_multiply);
+        break;
+    case ScatterUpdate::Reduction::MEAN :
+        scatterElementsUpdate<DT>(dstMemPtr, indicesMemPtr, updateMemPtr, axis, reduce_mean);
+        break;
+    default :
+        OPENVINO_THROW("unsupported reduce");
+        break;
+    }
+}
+
+void ScatterUpdate::scatterElementsUpdate(const MemoryPtr& dstMemPtr, const MemoryPtr& indicesMemPtr, const MemoryPtr& updateMemPtr, int axis) {
+    ScatterElementsUpdateContext ctx{this, dstMemPtr, indicesMemPtr, updateMemPtr, axis};
+    OV_SWITCH(intel_cpu,
+              ScatterElementsUpdateDispatcher,
+              ctx,
+              dataPrec,
+              OV_CASE(ov::element::f32, float),
+              OV_CASE(ov::element::i32, int32_t),
+              OV_CASE(ov::element::bf16, ov::bfloat16),
+              OV_CASE(ov::element::f16, ov::float16));
```

Comment:
Since we don't support u8/i8 types one it might become an issue for quantized models

Reply:
We can extend to support any precision, no problem.